pingcap/tidb-operator
# Fix tidb-init job incorrect check of %% for permit_host

### What problem does this PR solve? 
Fix #3290
### What is changed and how does it work?

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 1. set .Values.tidb.permitHost: '%' in tidb-cluser chart, helm install tidb-cluster
 2. helm install tidb-cluster -n tidb-cluster charts/tidb-cluster
 3. use mysql client connect to tidb server port and execute `select Host from mysql.user where User='root';`
Expected output:
```
+------+
| Host |
+------+
| %    |
+------+
1 row in set (0.00 sec)
```

Code changes

 - Has helm scripts change

Side effects

 None

Related changes

 - Need to cherry-pick to the release branch


### Does this PR introduce a user-facing change?:

```release-note
NONE
```
